publication-commons/src/test/java/cucumber/permissions/file/FileAccessFeatures.java

```diff
@@ -32,38 +32,42 @@ public FileAccessFeatures(FileScenarioContext scenarioContext) {
         scenarioContext.setResource(Resource.fromPublication(randomNonDegreePublication()));
-        scenarioContext.setFile(FileEntry.create(file, SortableIdentifier.next(), UserInstance.create(randomString(),
-                                                                                                      randomUri())));
+        scenarioContext.setFile(createFile(string, EMPTY_PROPERTY));
     }
 
     @Given("a file of type {string} with property {string}")
     public void aFileOfTheTypeAndWithProperty(String fileType, String fileProperty) throws ClassNotFoundException {
-        var file = createFile(fileType, fileProperty);
         if (fileProperty.toLowerCase().contains("degree")) {
             scenarioContext.setResource(Resource.fromPublication(randomDegreePublication()));
         } else {
             scenarioContext.setResource(Resource.fromPublication(randomNonDegreePublication()));
         }
-        scenarioContext.setFile(FileEntry.create(file, SortableIdentifier.next(), UserInstance.create(randomString(),
-                                                                                                         randomUri())));
-    }
 
-    @SuppressWarnings("unchecked")
-    private static File createFile(String fileType, String fileProperty) throws ClassNotFoundException {
-        var clazz = (Class<File>) Class.forName(File.class.getPackageName() + "." + fileType);
-        var file = File.builder();
-        if (fileProperty.toLowerCase().contains("embargo")) {
-            file.withEmbargoDate(Instant.now().plus(100, DAYS));
-        }
-        return file.build(clazz);
+        scenarioContext.setFile(createFile(fileType, fileProperty));
     }
 
     @When("the user have the role {string}")
     public void theUserHaveTheRole(String useraRole) {
-        var user = UserInstance.create(randomString(), randomUri());
-        scenarioContext.setUser(user);
+        if (useraRole.toLowerCase().contains("curator")) {
+            scenarioContext.getUserContext().accessRights.add(AccessRight.MANAGE_RESOURCES_STANDARD);
+        }
+        if (useraRole.toLowerCase().contains("file curator")) {
+            scenarioContext.getUserContext().accessRights.add(AccessRight.MANAGE_RESOURCE_FILES);
+        }
+        if (useraRole.toLowerCase().contains("curator at x") ||
+            useraRole.toLowerCase().contains("curators for other contributors")) {
+            scenarioContext.getResource()
+                .setCuratingInstitutions(
+                    Set.of(new CuratingInstitution(scenarioContext.getUserContext().topLevelOrgCristinId
+                        , Collections.emptySet())));
+        }
+        if (useraRole.toLowerCase().contains("curator at x")) {
+            scenarioContext.setFileOwnership(FileOwnership.OWNER);
+        }
+        if (useraRole.toLowerCase().contains("curators for other contributors")) {
+            scenarioContext.setFileOwnership(FileOwnership.NOT_OWNER);
+        }
```

Comment:
This is a bit hard to understand

Reply:
haha yes, would like to have a cleaning session in the test setup if you have time